azurerm/resource_arm_cognitive_account_test.go

```diff
@@ -0,0 +1,115 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	// "github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+// TODO: an update test going from External -> Internal -> External
+
+func TestAccAzureRMCognitiveAccount_basic(t *testing.T) {
+	log.SetOutput(os.Stdout)
+	resourceName := "azurerm_cognitive_account.test"
+	ri := acctest.RandInt()
+	config := testAccAzureRMCognitiveAccount_basic(ri, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMAppCognitiveAccountDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMCognitiveAccountExists(resourceName),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMAppCognitiveAccountDestroy(s *terraform.State) error {
+	// client := testAccProvider.Meta().(*ArmClient).cognitiveAccountsClient
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_cognitive_account" {
+			continue
+		}
+
+		// name := rs.Primary.Attributes["name"]
+		// resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		// ctx := testAccProvider.Meta().(*ArmClient).StopContext
+		/*
+			resp, err := client.Get(ctx, resourceGroup, name)
+
+			if err != nil {
+				if utils.ResponseWasNotFound(resp.Response) {
+					return nil
+				}
+
+				return err
+			}
+		*/
```

Comment:
can we add this check back in? This is used to ensure the Cognitive Services account has been deleted